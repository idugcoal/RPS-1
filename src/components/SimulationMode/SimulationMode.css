.counterContainer {
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin-bottom: 50px;
}

.reset {
  cursor: pointer;
}
