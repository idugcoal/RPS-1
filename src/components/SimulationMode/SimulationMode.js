import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Counter from '../Counter/Counter';
import Battlefield from '../Battlefield/Battlefield';
import { simulateComputer, computeWinner } from '../../utils';
import style from './SimulationMode.css';

export default class SimulationMode extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstWins: 0,
      secondWins: 0,
      firstSelection: undefined,
      secondSelection: undefined,
    };
  }

  componentDidMount() {
    setTimeout(() => this.simulateGame(), 1000);
  }

  componentWillUpdate() {
    setTimeout(() => this.simulateGame(), 1000);
  }

  simulateGame() {
    const firstSelection = simulateComputer();
    const secondSelection = simulateComputer();
    const result = computeWinner(firstSelection, secondSelection);
    this.setState(({ firstWins, secondWins }) => ({
      firstSelection,
      secondSelection,
      firstWins: result === 1 ? firstWins + 1 : firstWins,
      secondWins: result === -1 ? secondWins + 1 : secondWins,
    }));
  }

  render() {
    const {
      firstWins,
      secondWins,
      firstSelection,
      secondSelection,
    } = this.state;
    const { onToggle } = this.props;

    return (
      <div>
        <div className={style.counterContainer}>
          <Counter name='Cpu 1' wins={firstWins} />
          <div
            className={style.reset}
            onClick={onToggle}
          >
            Player Mode
          </div>
          <Counter name='Cpu 2' wins={secondWins} />
        </div>
        <Battlefield
          firstName='Cpu 1'
          secondName='Cpu 2'
          firstSelection={firstSelection}
          secondSelection={secondSelection}
        />
      </div>
    );
  }
}

SimulationMode.propTypes = {
  onToggle: PropTypes.func,
};
