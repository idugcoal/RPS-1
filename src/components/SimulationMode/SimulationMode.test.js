import React from 'react';
import { shallow } from 'enzyme';
import SimulationMode from './SimulationMode';

test('should render correctly', () => {
  const wrapper = shallow(<SimulationMode />);
  expect(wrapper).toMatchSnapshot();
});
