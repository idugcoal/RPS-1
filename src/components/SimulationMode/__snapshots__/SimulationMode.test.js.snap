// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div>
    <ComputerCounter
      name="Cpu 1"
      wins={0}
    />
    <div>
      Player Mode
    </div>
    <ComputerCounter
      name="Cpu 2"
      wins={0}
    />
  </div>
  <Battlefield
    firstName="Cpu 1"
    secondName="Cpu 2"
  />
</div>
`;
